### PR TITLE
feat: allow user to auth with env vars on cli

### PR DESCRIPTION
**Pull request by owlas**
Closes: #2495

### CODE REVIEW

**Comment by rephus** [comment]
I can't changed the project via env variable 

image

File: packages/cli/src/config.ts [hunk 1]
```diff
@@ -14,17 +14,30 @@ type Config = {
...
+        context: {
+            ...(validated.context || {}),
+            apiKey: process.env.LIGHTDASH_API_KEY || validated.context?.apiKey,
+            project:
+                process.env.LIGHTDASH_PROJECT || validated.context?.project,
+            serverUrl:
+                process.env.LIGHTDASH_URL || validated.context?.serverUrl,
+        },
+    };
+    return useEnv ? envOverrides : validated;
```
[the remark this answers is not in the document]
Comment (owlas) [reply packages/cli/src/config.ts:40]: I'm not sure I completely understand. What do you mean by "won't get any of the env values" ?

[the remark this answers is not in the document]
Comment (ZeRego) [reply packages/cli/src/config.ts:40]: In the login command you changed to `getConfig(false)` so it won't get the value of LIGHTDASH_PROJECT. 
In the set project command you changed to `getConfig(false)` so it won't get the value of LIGHTDASH_API_KEY and LIGHTDASH_URL.

I wasn't thinking about this logic from a cli perspective where each command needs the env vars. So even if these commands don't support it, the user would still use the env vars in the other commands and they would use the env vars. 👍

**Comment by owlas** [comment]
@rephus what if you put `export` before the env var? Or do it on the same line?

**Review by rephus — APPROVED** [review]
It works with `export` 

image

**Pull request merged**